opensir/models/model.py

```diff
@@ -145,12 +147,20 @@ def function_handle(t, *par_fit, pop=population):
             params = np.array(self.p)
             params[fit_index] = par_fit
             self.p = params
-            i_mod = call_solver(self._model, self.p, self.w0, t)
-            return i_mod[:, 2] * pop
+            # SIR-X special fitting of initial conditions
+            if self.name == "sirx":
```

Comment:
All children specific code should be in their respective classes, not in the parent class.
This pattern should be modelled with e.g a "get_initial_conds()" function implements by both children classes (SIR and SIR-X)